data-model/annotations/0.1.0/schema/annotation-event-for-web.json

```diff
@@ -0,0 +1,189 @@
+                                "properties": {
+                                  "ac:xFrac": {
+                                    "type": "number",
+                                    "minimum": 0,
+                                    "maximum": 1
+                                  },
+                                  "ac:yFrac": {
+                                    "type": "number",
+                                    "minimum": 0,
+                                    "maximum": 1
+                                  },
+                                  "ac:widthFrac": {
+                                    "type": "number",
+                                    "minimum": 0,
+                                    "maximum": 1
+                                  },
+                                  "ac:heightFrac": {
+                                    "type": "number",
+                                    "minimum": 0,
+                                    "maximum": 1
+                                  }
+                                },
+                                "required": [
+                                  "ac:xFrac",
+                                  "ac:yFrac",
+                                  "ac:widthFrac",
+                                  "ac:heightFrac"
+                                ]
+                              },
+                              "dcterms:conformsTo": {
+                                "type": "string"
+                              }
+                            },
+                            "required": [
+                              "ods:type",
+                              "ac:hasRoi"
+                            ]
+                          }
+                        ]
+                      }
+                    },
+                    "required": [
+                      "ods:id",
+                      "ods:type"
+                    ]
+                  },
+                  "oa:body": {
+                    "type": "object",
+                    "properties": {
+                      "ods:type": {
+                        "type": "string"
+                      },
+                      "oa:value": {
+                        "type": "array"
+                      },
+                      "dcterms:reference": {
+                        "type": "string"
+                      },
+                      "ods:score": {
+                        "type": "number",
+                        "minimum": 0,
+                        "maximum": 1
+                      }
+                    },
+                    "required": [
+                      "ods:type",
+                      "oa:value"
+                    ]
+                  }
+                },
+                "required": [
+                  "oa:motivation",
+                  "oa:target",
+                  "oa:body"
+                ]
+              },
+              "minItems": 1,
+              "maxItems": 1
```

Comment:
So there is only one annotation allowed per request?
What if there is MAS with annotations on multiple fields?

Reply:
This schema is specifically for requesting batch annotations from the Front End. you only send one annotation to the backend from DiSSCover.

[Event for MAS](https://github.com/DiSSCo/openDS/blob/master/data-model/annotations/0.1.0/schema/annotation-event.json). 

MASs are allowed to send multiple annotations, but the front end is not. (That could change, but we don't currently support it). 

batching works like:
1. **front end sends single annotation** and batch metadata, which defines the parameters for what other annotations to batch
1a. backend searches the parameters, returns total number of objects that would be annotated according to the batch metadata
1b. user confirms that number of objects should be annotated
2. processing service finds the annotated objects that meet that criteria, build annotations, then sends it back to itself to process the new batch result annotations

so the FE will only send one annotation, even in a batch request

Reply:
Sounds like a good approach!